### PR TITLE
User should be able to search across the web

### PR DESCRIPTION
**Pull request by Favor-star**
## Purpose

This PR is intended to make the user be able to search across the websites  through search input in navbar

## Changes Made

Created search pages including search header and search results components 

## Testing Instructions

No tests made yet

## Related Issues

THe issues is with redux. Data seems not stored in redux. which requires me to re-request data from backend.

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:

- [x] Code follows the project's coding standards
- [x] Changes are covered by tests
- [x] Documentation is updated (if applicable)
- [x] All tests pass
Screenshot 2024-06-24 131841

### CODE REVIEW

File: src/components/search/SearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const SearchResults = ({ searchQuery }: SearchQueryProps) => {
+  const { data, error, isLoading } = useGetProductsQuery();
+  if (error) {
+    console.log(error);
+    return;
+  }
+  const productsData: ProductResponse = data as unknown as ProductResponse;
+  const productsList: Product[] = productsData ? productsData.data : [];
+
+  console.log(productsList);
```
Comment (patrickhag) [inline src/components/search/SearchResults.tsx]: can you remove console.logs?

File: src/components/search/SearchResults.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  );
+
+  console.log(filteredProducts);
+  // const products = useSelector((state: any) => state.products.productsDataList);
+  // const productFetched = useSelector((state: any) => state.products.productFetched);
+
+  // useEffect(() => {
+  //   console.log(productFetched);
+  //   console.log(products);
+  // }, [productFetched, products]);
```
Comment (patrickhag) [inline src/components/search/SearchResults.tsx]: why's this code commented?

**Review by patrickhag — CHANGES_REQUESTED** [review]
consider these comments!

**Review by gracemugwanezak — APPROVED** [review]

File: src/components/search/SearchResults.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const { data, error, isLoading } = useGetProductsQuery();
+  if (error) {
+   
+    return;
+  }
+  const productsData: ProductResponse = data as unknown as ProductResponse;
+  const productsList: Product[] = productsData ? productsData.data : [];
+
+  const filteredProducts = productsList.filter(product =>
+    product.name.toLowerCase().includes(searchQuery.toLowerCase())
```
Comment (Mag-codes) [inline src/components/search/SearchResults.tsx]: image

include more than  search by product name

File: src/components/search/SearchResults.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  };
+  return (
+    <div className='p-3 md:p-4 xl:px-10 flex flex-row flex-wrap gap-4'>
+     
+      {isLoading ? (
+        'Loading....'
+      ) : filteredProducts.length === 0 ? (
+        <div className='w-full '>
+          <p className='font-medium text-2xl italic'>
+            No matching products were found for <span className='font-bold'>{searchQuery}</span>
```
Comment (Mag-codes) [inline src/components/search/SearchResults.tsx]: image

Show available products as a user types in a overlay

**Review by patrickhag — APPROVED** [review]
LGTM

**Review by Mag-codes — CHANGES_REQUESTED** [review]

**Review by hozayves — APPROVED** [review]

**Review by P-Rwirangira — APPROVED** [review]

**Pull request closed without merging**